.github/workflows/ci.yaml

```diff
@@ -0,0 +1,31 @@
+name: Build and Test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
+        go: [1.13, 1.12]
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-go@v1
+      with:
+        go-version: ${{ matrix.go }}
+    - name: Setup env
+      run: |
+        # https://github.com/actions/setup-go/issues/14
+        echo "::set-env name=GOPATH::$(go env GOPATH)"
+        echo "::add-path::$(go env GOPATH)/bin"
+      shell: bash
+    - name: Install Dependencies
+      run: |
+        sudo apt-get install libnl-3-dev libnl-genl-3-dev
+    - name: Build
+      run: |
+        go build ./...
```

Comment:
how is GO111MODULE set? looks like if it's go1.12, it needs pull in all the deps first?

Reply:
It is not set. For go 1.12, by default it is `auto` mode, hence it'll use module mode since the project is checked out outside of GOPATH.

Anyway, it'll need to download modules first, same case for 1.12 or 1.13. I think it probably can be further optimized by enabling action/cache, but prefer to leave out the complexity for now.